### DISCUSSION TITLE
docker-compose; bug for password with special characters

### DISCUSSION THREAD

**Discussion by Brtrnd** [photoprism/photoprism#2094] [Q&A] [answered]
Hi
I created this thread on reddit and I think it's a bug. But I'm not sure so here I am.
https://www.reddit.com/r/photoprism/comments/t3fmwb/networking_issues/hyw86ph/?context=3 

Someone told me to remove special characters out of the password and suddenly it worked.
To be honest, I'm not very active on github, but I'll try to followup.

copy of the message on reddit:
Hi

I have an issue I'm not sure is 100% a photoprism issue.
I tried installing photoprism on my Synology Nas using docker compose.
When running with SQLlite, it starts without an issue.
When trying to connect to my existing mariaDB instance; I get an error message `Error 1045: Access denied for user 'photoprism'@'[172.18.0.1](https://172.18.0.1/)' (using password: YES)`

If I start photoprism with SQLlite; I go in the container and run `mysql -h [10.0.2.5](https://10.0.2.5/) -u photoprism -p` it will connect to the database without an issue.
From my own pc I can connect to the database server and select the photoprism db.

From this I understand it's not a network issue; but rather a configuration issue.
My database is an empty database, without tables. created according to [the database docs](https://docs.photoprism.app/getting-started/advanced/databases/#mariadb).
I'm also certain the user/pass is correct.

If relevant, I uploaded [my yaml file here](https://pastebin.com/9akvUq2w).
A screenshot with photoprism environment _DEBUG:"true":
![screenshot](https://preview.redd.it/w9sclm8nrkk81.png?width=1516&format=png&auto=webp&s=06830524826d11c418821614a1cc100426763a0e)

**Comment by lastzero** [comment, marked as answer]
$ must be escaped as $$ so it doesn't get interpreted as variable
